test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -752,6 +754,128 @@ TEST(ParticipantTests, GetParticipantProfileQos)
+    DomainParticipantFactory::get_instance()->load_XML_profiles_file(xml_filename);
+    // NOTE: At the time of this writing, the only way to reset the default qos after loading an XML is to do as follows
+    DomainParticipantFactory::get_instance()->load_profiles();
+    DomainParticipantFactory::get_instance()->set_default_participant_qos(PARTICIPANT_QOS_DEFAULT);
+    DomainParticipantQos default_qos_from_profile;
+    EXPECT_EQ(
+        DomainParticipantFactory::get_instance()->get_default_participant_qos(default_qos_from_profile),
+        RETCODE_OK);
+
+    // Check they correspond to the same profile
+    EXPECT_EQ(default_qos, default_qos_from_profile);
+}
+
+TEST(ParticipantTests, GetParticipantExtendedQosFromXml)
+{
+    const std::string xml_filename("test_xml_profile.xml");
+    const std::string profile_name("test_participant_profile");
+
+    std::string complete_xml = testing::load_file(xml_filename);
+
+    // Get QoS given profile name
+    DomainParticipantExtendedQos qos;
+    EXPECT_EQ(
+        DomainParticipantFactory::get_instance()->get_participant_extended_qos_from_xml(complete_xml, qos,
+        profile_name),
+        RETCODE_OK);
+
+    // Get QoS without providing profile name (gets first one found)
+    DomainParticipantExtendedQos qos_empty_profile;
+    EXPECT_EQ(
+        DomainParticipantFactory::get_instance()->get_participant_extended_qos_from_xml(complete_xml, qos_empty_profile), RETCODE_OK);
+
+    // Check they correspond to the same profile
+    // NOTE: test_participant_profile is assumed to be the first participant profile in the XML file
+    EXPECT_EQ(qos, qos_empty_profile);
+
+    // Load profiles from XML file and get QoS given profile name
+    DomainParticipantFactory::get_instance()->load_XML_profiles_file(xml_filename);
+    DomainParticipantExtendedQos qos_from_profile;
+    EXPECT_EQ(
+        DomainParticipantFactory::get_instance()->get_participant_extended_qos_from_profile(profile_name,
+        qos_from_profile),
+        RETCODE_OK);
+
+    // Check they correspond to the same profile
+    EXPECT_EQ(qos, qos_from_profile);
+
+    // Test return when a non-existent profile is used
+    EXPECT_EQ(
+        DomainParticipantFactory::get_instance()->get_participant_extended_qos_from_xml(complete_xml, qos,
+        "incorrect_profile_name"),
+        RETCODE_BAD_PARAMETER);
+}
+
+TEST(ParticipantTests, GetDefaultParticipantExtendedQosFromXml)
+{
+    const std::string xml_filename("test_xml_profile.xml");
+
+    std::string complete_xml = testing::load_file(xml_filename);
+
+    // Get default QoS from XML
+    DomainParticipantExtendedQos default_qos;
+    EXPECT_EQ(
+        DomainParticipantFactory::get_instance()->get_default_participant_extended_qos_from_xml(complete_xml, default_qos),
+        RETCODE_OK);
+
+    // NOTE: cannot load profiles file and compare with default value as
+    // DomainParticipantFactory::get_default_participant_extended_qos is currently unavailable
```

Comment:
But you can `load_XML_profiles_file()` and retrieve the `"test_participant_profile"` which is the default one and compare with it.

```
// Load profiles from XML file and get QoS given profile name
    DomainParticipantFactory::get_instance()->load_XML_profiles_file(xml_filename);
    DomainParticipantExtendedQos qos_from_profile;
    EXPECT_EQ(
        DomainParticipantFactory::get_instance()->get_participant_extended_qos_from_profile("test_participant_profile",
        qos_from_profile),
        RETCODE_OK);

    // Check they correspond to the same profile
    EXPECT_EQ(qos, qos_from_profile);
```